client/src/Components/RegisterPage/Stepper/Steps/DetailsComponent.jsx

```diff
@@ -150,17 +153,22 @@ export default function StepThreeComponent({ detailsForm }) {
           الرجوع للخلف
         </Button>
 
-        <Button
+        <LoadingButton
+          loading={loading}
+          onClick={() => {
+            setLoading(true);
+            detailsForm.handleSubmit();
```

Comment:
So, if there's an error, loading will be true, So user won't be able to register  again,
I'd prefer to remove the loading state to the place you declare `onFinish` function,
And set loading to true before sending request and then to false in case of response error or success